instrumentation/pulsar/pulsar-2.8/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/pulsar/v2_8/PulsarClientTest.java

```diff
@@ -154,26 +71,35 @@ void testConsumeNonPartitionedTopic() throws Exception {
 
     latch.await(1, TimeUnit.MINUTES);
 
-    testing.waitAndAssertTraces(
+    AtomicReference<SpanData> producerSpan = new AtomicReference<>();
+    testing.waitAndAssertSortedTraces(
```

Comment:
Why are we using `waitAndAssertSortedTraces` here but just `waitAndAssertTraces` in the suppressed receive test? Can we use the sorted in both places for consistency?

Reply:
We don't need to use `waitAndAssertSortedTraces` in the suppress tests because they usually only produce a single trace so there aren't ordering issues.